src/server.c

```diff
@@ -7212,6 +7212,11 @@ int main(int argc, char **argv) {
         loadDataFromDisk();
         aofOpenIfNeededOnServerStart();
         aofDelHistoryFiles();
+        /* While loading data, we delay applying "appendonly" config change.
+         * If there was a config change while we were inside loadDataFromDisk()
+         * above, we'll apply it here. */
+        applyAppendOnlyConfig();
```

Comment:
i think we need to look at the return value and do something similar to restartAOFAfterSYNC (i.e. `exit(1)`).
when we do that as part of the CONFIG command we can return an error indication to the user, but if we do that later, and we fail applying the config, then we can't ignore that error.

Reply:
added retry logic